org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -101,16 +105,40 @@ public boolean visit(IResourceDelta delta) throws CoreException {
 		if (resource.getType() == IResource.FOLDER || resource.getType() == IResource.ROOT) {
 			return true;
 		}
-
 		if (resource.getType() == IResource.PROJECT) {
 			// ignore problems caused by standalone files (problems in the default project)
 			if (JavaLanguageServerPlugin.getProjectsManager().getDefaultProject().equals(resource.getProject())) {
 				return false;
 			}
 			// report problems for other projects
-			String uri = JDTUtils.getFileURI(resource);
 			IMarker[] markers = resource.findMarkers(null, true, IResource.DEPTH_ONE);
 			Range range = new Range(new Position(0, 0), new Position(0, 0));
+			List<IMarker> markerList = new ArrayList<>();
+			for (IMarker marker : markers) {
+				markerList.add(marker);
+			}
+			Iterator<IMarker> iter = markerList.iterator();
+			while (iter.hasNext()) {
+				IMarker marker = iter.next();
+				if (!marker.exists()) {
+					iter.remove();
+					continue;
+				}
+				if (IMavenConstants.MARKER_CONFIGURATION_ID.equals(marker.getType())) {
+					String message = marker.getAttribute(IMarker.MESSAGE, "");
+					if (message != null && message.equals(Messages.ProjectConfigurationUpdateRequired)) {
```

Comment:
Messages.ProjectConfigurationUpdateRequired.equals(message), no need for null-check. It can't be null anyways